pkgs/applications/emulators/duckstation/default.nix

```diff
@@ -110,14 +89,28 @@ stdenv.mkDerivation (finalAttrs: {
     cp -r bin $out/share/duckstation
     ln -s $out/share/duckstation/duckstation-qt $out/bin/
 
-    install -Dm644 bin/resources/images/duck.png $out/share/pixmaps/duckstation.png
+    install -Dm644 $src/scripts/org.duckstation.DuckStation.desktop $out/share/applications/org.duckstation.DuckStation.desktop
+    install -Dm644 $src/scripts/org.duckstation.DuckStation.png $out/share/pixmaps/org.duckstation.DuckStation.png
 
     runHook postInstall
   '';
 
-  qtWrapperArgs = [
-    "--prefix LD_LIBRARY_PATH : ${lib.makeLibraryPath ([ vulkan-loader ] ++ cubeb.passthru.backendLibs)}"
-  ];
+  qtWrapperArgs =
+    let
+      libPath = lib.makeLibraryPath ([
+        vulkan-loader
+      ] ++ cubeb.passthru.backendLibs);
+    in [
+      "--prefix LD_LIBRARY_PATH : ${libPath}"
+    ];
+
+  # https://github.com/stenzek/duckstation/blob/master/scripts/appimage/apprun-hooks/default-to-x11.sh
+  # Can't avoid the double wrapping, the binary wrapper from qtWrapperArgs doesn't support --run
```

Comment:
Doing this would increase startup time, though.
So... which one is less bad? Double wrapping, or longer startup time?
(Also, if I do the latter, it should be changed on PCSX2 too for consistency)

Reply:
I prefer no double-wrapping.

How much the startup would be affected?

Reply:
See #225881.
(Looking at the that... how am I supposed to revert it? Overriding `propagatedBuildInputs`?)

Reply:
The speedup is huge, two orders of magnitude! Let's keep the double wrapping, until someone writes the Best Wrappers Ever (using JSON or other blue magic).

Reply:
Double wrapping is really bad as the program thinks that it was started as .name-wrapped which really is the name of the first wrapper after the other one was applied on top. 

Also instead of adding a new magic variable which does things, why can't we tell people to just export QT_QPA_PLATFORM=xcb for duckstation? That would avoid the problem entirely.

Reply:
So, I'll reiterate the points:
- This wrapping is done to match upstream's behavior. The AppImages have Wayland as an opt-in (and not opt-out) because they think that the Wayland experience on PCSX2/DuckStation isn't good enough for it to be the default. See `https://github.com/PCSX2/pcsx2/pull/10179`. It is important to respect this when making a package of their program.
- So, if I have to do the above while not double wrapping, then I need to "revert" #225881's behavior (that is, make `wrapQtAppsHook` use `makeWrapper` instead of `makeBinaryWrapper`). I don't know how to do that.

<sub><sup>Also, no one said anything [when I did the same thing on PCSX2](https://github.com/NixOS/nixpkgs/pull/273330).</sup></sub>

Reply:
I read the small words.
And, seriously, the "no one said this to me before" is a bad argument, especially in a codebase with sudden changes like Nixpkgs.

> Also instead of adding a new magic variable which does things, why can't we tell people to just export QT_QPA_PLATFORM=xcb for duckstation? That would avoid the problem entirely.

I think it would be a serious problem of usability. Yes, we Nixpkgs users are usually a well-informed people, however things like "extra preparation steps" are too much to require, since

1. Other distros rarely worry about those things
2. export variables can affect more than the running program, breaking one of the main flagships of Nixpkgs - the sandbox
3. it is easy to forget to undo the exports, or even why they were made in the first place

Reply:
Do you consider this thread blocking? Would a simple `s/I_WANT_A_BROKEN_WAYLAND_UI/DUCKSTATION_I_WANT_A_BROKEN_WAYLAND_UI/` suffice?

Reply:
As the maintainer of this package, everything LGTM.

The problems of this expression are external to it.